client/src/components/geneExpression/index.tsx

```diff
@@ -53,39 +55,54 @@ class GeneExpression extends React.Component<{}, State> {
   }
 
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  renderGeneSets = () => {
+  renderGeneSets = (getMarkerGeneSets: boolean) => {
     const sets = [];
     // @ts-expect-error ts-migrate(2339) FIXME: Property 'genesets' does not exist on type 'Readon... Remove this comment to see the full error message
-    const { genesets } = this.props;
+    const { genesets, isCellGuideCxg } = this.props;
     const { geneIds, geneNames } = this.state;
 
     for (const [name, geneset] of genesets) {
-      const genesetIds = [];
-      const genesetNames = [];
-
-      // find ensembl IDs for each gene in the geneset
-      for (const gene of geneset.genes) {
-        let geneId;
-        if (geneIds) {
-          geneId = geneIds[geneNames.indexOf(gene[0])];
-        } else {
-          geneId = "";
+      if (
+        (!name.includes(" - marker genes") &&
```

Comment:
It's gated by the isCellGuideCxg, but they could indeed create a geneset in those instances with - marker genes at the end... I don't think it's that bad for it to move into the marker gene sets header in that context. It won't affect other explorer instances. What do you think?

Reply:
feature! not bug

Reply:
obviously I haven't read the PR yet, but is `isCellGuideCxg` a temp feature flag? If yes, then once we remove the feature flag what will happen?

Reply:
can we extract the constant string and reuse it to keep the code DRY please 😁 🙏

Reply:
It's not a temporary feature flag because CellGuide CXGs will always behave differently than nonCellGuide CXGs, but the places that are GATED by this flag can change

Reply:
Made it DRY!